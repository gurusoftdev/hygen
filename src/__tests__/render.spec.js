const render = require('../render')
const path = require('path')
const ftest = require('../test/ftest')
const fs = require('fs')
const fixture = name =>
  fs.readFileSync(path.join(__dirname, './fixtures', name)).toString()

describe('render', () => {
  ftest(
    'empty',
    { app: { action: { 'empty.ejs.t': fixture('empty.ejs.t') } } },
    () => {
      const res = render({})('app/action')
      console.log(res[0].file)
      expect(res[0].file).toMatch(/empty/)
      res[0].file = 'empty.ejs.t'
      expect(res).toMatchSnapshot()
    }
  )
  ftest(
    'full',
    { app: { action: { 'full.ejs.t': fixture('full.ejs.t') } } },
    () => {
      const res = render({ bill: 17, name: 'someone' })('app/action')
      console.log(res[0].file)
      expect(res[0].file).toMatch(/full/)
      res[0].file = 'full.ejs.t'
      expect(res).toMatchSnapshot()
    }
  )
})
