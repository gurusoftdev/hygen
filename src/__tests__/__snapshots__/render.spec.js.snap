// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render empty 1`] = `
Array [
  Object {
    "attributes": Object {},
    "body": "",
    "file": "empty.ejs.t",
  },
]
`;

exports[`render full 1`] = `
Array [
  Object {
    "attributes": Object {
      "to": "foo/someone/bar",
    },
    "body": "

This is the html email template.
Find me at <i>app/mailers/hello/html.ejs</i>

<br/> 
<br/> 

You owe
17",
    "file": "full.ejs.t",
  },
]
`;
